node/stakeValuesProcessor/stakeValuesProcessor.go

```diff
@@ -80,29 +102,47 @@ func (svp *stakedValuesProc) computeStakedValueAndTopUp() (*big.Int, *big.Int, e
 		return nil, nil, err
 	}
 
-	numRegistedNodes := uint64(0)
-	totalStakedValueAllLeaves := big.NewInt(0)
+	totalStaked, totalTopUp := big.NewInt(0), big.NewInt(0)
 	for leaf := range chLeaves {
-		validatorData := &systemSmartContracts.ValidatorDataV2{}
-		value, errTrim := leaf.ValueWithoutSuffix(append(leaf.Key(), vm.ValidatorSCAddress...))
-		if errTrim != nil {
-			return nil, nil, fmt.Errorf("%w for validator key %s", errTrim, hex.EncodeToString(leaf.Key()))
+		if len(leaf.Key()) != svp.publicKeyConverter.Len() {
```

Comment:
Slight optimization: could have "cached" the `svp.publicKeyConverter.Len()` in a variable on L105.

Reply:
done.

Reply:
fair enough. You cached the leafKey :D

Reply:
:+1: